src/hobbits-core/bitarray.cpp

```diff
@@ -771,6 +771,122 @@ QSharedPointer<BitArray> BitArray::fromString(QString bitArraySpec, QStringList
     }
 }
 
+char BitArray::hexTable(qint64 nibble) const {
+    char chr;
+    if (nibble == 0) {
+        chr = '0';
+    } else if (nibble == 1) {
+        chr = '1';
+    } else if (nibble == 2) {
+        chr = '2';
+    } else if (nibble == 3) {
+        chr = '3';
+    } else if (nibble == 4) {
+        chr = '4';
+    } else if (nibble == 5) {
+        chr = '5';
+    } else if (nibble == 6) {
+        chr = '6';
+    } else if (nibble == 7) {
+        chr = '7';
+    } else if (nibble == 8) {
+        chr = '8';
+    } else if (nibble == 9) {
+        chr = '9';
+    } else if (nibble == 10) {
+        chr = 'a';
+    } else if (nibble == 11) {
+        chr = 'b';
+    } else if (nibble == 12) {
+        chr = 'c';
+    } else if (nibble == 13) {
+        chr = 'd';
+    } else if (nibble == 14) {
+        chr = 'e';
+    } else if (nibble == 15) {
+        chr = 'f';
+    } else {
+        chr = 'o';
+    }
+    return chr;
+}
+
+QString BitArray::toBin() const {
+    QString str = "";
+
+    for (quint64 i = 0; i < this->sizeInBits(); i ++) {
+        if (i < this->sizeInBits() && i >= 0) {
+            if (this->at(i) == false) {
+                str+="0";
+            } else {
+                str+="1";
+            }
+        }
+    }
+    return str;
+}
+
+QString BitArray::toBin(qint64 start, int length) const {
+    QString str = "";
+
+    for (quint64 i = start; i < start + length; i ++) {
```

Comment:
Instead of checking this condition at every step of the loop, you could probably check it once before the loop and maybe adjust the numbers to make sure length stays in bounds.